server/src/main/java/org/elasticsearch/search/query/QueryPhase.java

```diff
@@ -265,9 +280,8 @@ static boolean executeInternal(SearchContext searchContext) throws QueryPhaseExe
             }
 
             if (searchContext.lowLevelCancellation()) {
-                SearchShardTask task = searchContext.getTask();
```

Comment:
I have a slight preference for this version. The `task` should never change during the execution of a search request.